src/rust-context.h

```diff
@@ -22,11 +22,13 @@
 #include "detect.h"
 #include "detect-engine-state.h" //DetectEngineState
 
-#include "app-layer-krb5.h" //KRB5State, KRB5Transaction
 #include "app-layer-ike.h"  //IKEState, IKETransaction
-#include "app-layer-ntp.h" //NTPState, NTPTransaction
-#include "app-layer-snmp.h" //SNMPState, SNMPTransaction
 #include "app-layer-tftp.h" //TFTPState, TFTPTransaction
+/** Opaque Rust types. */
+typedef struct KRB5State_ KRB5State;
+typedef struct KRB5Transaction_ KRB5Transaction;
+typedef struct SNMPState_ SNMPState;
+typedef struct SNMPTransaction_ SNMPTransaction;
```

Comment:
@jasonish I gather that the current code has 3 solutions for this
- rust/cbindgen.toml having an exclusion like "DNSTransaction" = "RSDNSTransaction"
- Do not use a direct rust transaction type in a C api function, but just a pointer, and do the unsafe cast in rust
- this kind of SNMPTransaction opaque type in rust-context.h

Am I correct ?
Is one solution better ?

Reply:
I do not think that cbindgen can create complex structs with `Vec` and such, can it ?

Reply:
It'll create your opaque structs for you, ie)
```
typedef struct SNMPTransaction SNMPTransaction;
```